packages/utils/src/matchDirective.ts

```diff
@@ -0,0 +1,22 @@
+import { DirectiveNode, ArgumentNode, StringValueNode } from 'graphql';
+
+export function matchDirective(
+  directive: DirectiveNode,
+  requiredName: string,
+  requiredArgs: Record<string, any> = {}
+): boolean {
+  if (directive.name.value === requiredName) {
+    const requiredArgNames = Object.keys(requiredArgs);
+
+    if (requiredArgNames.length) {
+      const argsMap: Record<string, any> = Object.create(null);
+      directive.arguments.forEach((arg: ArgumentNode) => {
+        const argValue = arg.value as StringValueNode;
```

Comment:
@yaacovCR – is this an acceptable way to cheat the TypeScript compiler into allowing a `value` reference on `ValueNode`? I'm on about day-3 with typescript now.

Reply:
it's acceptable if you know the type of the directive, which you may, but you can also use `getDirectiveValues` fro graphql-js and `getDirectives` from @graphql-tools/utils

Reply:
Those are awkward here because there's no schema instance available in the constructor where the transforms are memoized. Looks like a `'value' in dir.value` check gets the job done without lying about it.